### PR TITLE
ISPN-16585 Apply max result on (projection-only-hybrid) queries

### PR DESCRIPTION
**Pull request by fax4ever**
https://issues.redhat.com/browse/ISPN-16585

Based on https://github.com/infinispan/infinispan/pull/12860 that should be merged first!

### CODE REVIEW

**Comment by fax4ever** [comment]
Rebased to fix the CI

**Comment by fax4ever** [comment]
I rebased this contribution on top of its base contribution (#12860) - aligning also the new client module.

**Review by wburns — APPROVED** [review]
LGTM as far as I can tell

File: commons/all/src/main/java/org/infinispan/commons/util/Closeables.java [hunk 1]
```diff
@@ -207,6 +217,19 @@ public void close() {
       }
    }
 
+   private static class StreamToCloseableIteratorWithCount<E> extends StreamToCloseableIterator<E>
```
[the remark this answers is not in the document]
Comment (fax4ever) [reply commons/all/src/main/java/org/infinispan/commons/util/Closeables.java:220]: you're right!

**Review by pruivo — APPROVED** [review]

**Comment by pruivo** [comment]
Failures are unrelated. Merged! Thanks @fax4ever

**Comment by fax4ever** [comment]
> Failures are unrelated. Merged! Thanks @fax4ever

Thanks @pruivo!

**Pull request merged**